### PR TITLE
List endpoint for cluster template instances

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: List endpoint for cluster template instances

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3464

```release-note
Templates for cluster creation
```

### CODE REVIEW

File: pkg/handler/v2/cluster_template/cluster_template_test.go [hunk 1]
```diff
@@ -577,3 +577,57 @@ func TestGetClusterTemplateInstance(t *testing.T) {
...
+		TemplateID             string
+		HTTPStatus             int
+		ExistingAPIUser        *apiv1.User
+		ExistingKubermaticObjs []ctrlruntimeclient.Object
+	}{
+		// scenario 1
+		{
+			Name:             "scenario 1: get template instance for global template",
+			TemplateID:       "ctID2",
+			ExpectedResponse: `[{"name":"","spec":{"projectID":"","clusterTemplateID":"","clusterTemplateName":"","replicas":0}},{"name":"my-first-project-ID-ctID2","spec":{"projectID":"my-first-project-ID","clusterTemplateID":"ctID2","clusterTemplateName":"","replicas":10}}]`,
```
Comment (lsviben) [inline pkg/handler/v2/cluster_template/cluster_template_test.go]: why is this empty instance in the response?

Comment (zreigz) [reply pkg/handler/v2/cluster_template/cluster_template_test.go]: good point. There were two elements in the list. I copy-paste the first empty element.
I fixed this

**Pull request merged**